core/src/main/scala/org/apache/spark/scheduler/BlacklistTracker.scala

```diff
@@ -336,9 +336,9 @@ private[scheduler] object BlacklistTracker extends Logging {
    */
   def isBlacklistEnabled(conf: SparkConf): Boolean = {
     conf.get(config.BLACKLIST_ENABLED) match {
-      case Some(enabled) =>
+      case enabled =>
         enabled
-      case None =>
+      case _ =>
```

Comment:
From my understanding, the code will never fall back to this block, so the following logic will never be executed. This is semantically different from previous code.